README.md

```diff
@@ -58,10 +64,42 @@ Don't start this too quickly after starting the flight controller and Unreal. MA
 
 To use PX4, add the arg `ardupilot:=false`. 
 
-The camera ROS publishers are disabled by default as it slows the publishing rate for everything to ~3Hz. To run with camera publishers on, add the arg `enable_cameras:=true`. When running with cameras enabled, Rviz looks like this.
+The camera ROS publishers are disabled by default as it slows the publishing rate for everything to ~3Hz. To run with camera publishers on, add the arg `enable_cameras:=true`. When running with cameras enabled and do_octomap set to true, Rviz looks like this.
 
 ![](images/airsim-start-ros.png)
 
+### manual flight
+
+To test how the map looks as the vehicle moves around, you can fly the ArduPilot drone with terminal commands. In the ArduPilot terminal, run:
+
+```
+mode guided
+arm throttle
+takeoff 2
+module load message
+```
+Then you can use any of the following commands.
+
+* message SET_POSITION_TARGET_LOCAL_NED 0 0 0 9 3576 20 0 0 0 0 0 0 0 0 0 0
+    * Flies 20m forward (necessary in Blocks to see an obstacle)
+* message SET_POSITION_TARGET_LOCAL_NED 0 0 0 9 3576 0 0 -2 0 0 0 0 0 0 0 0
+    * Flies up 5m
+* message SET_POSITION_TARGET_LOCAL_NED 0 0 0 9 2503 0 0 0 0 0 0 0 0 0 1.5707963 0
+    * Rotates 90 deg to right
+* message SET_POSITION_TARGET_OFFSET_NED 0 0 0 9 3527 0 0 0 2 0 0 0 0 0 0 0
+    * Forward velocity at 2 m/s for 3 seconds (built-in velocity command stops after 3 sec)
+
+
 ## notes
 
-This wrapper removed a lot of the abstractions of the original Microsoft ROS wrapper in favor of efficiency. The result is a wrapper that runs faster, but only for a single MultiRotor, and only when using one of the two main industry standard flight controllers (PX4 or ArduPilot). We also corrected some issues with left/right frames being flipped, static transforms published as dynamic and vice versa, and reverted to standard UAS frame conventions where needed for base_link/base_link_frd, map/map_ned, and odom/odom_ned.
\ No newline at end of file
+This wrapper removed a lot of the abstractions of the original Microsoft ROS wrapper in favor of efficiency. The result is a wrapper that runs faster, but only for a single MultiRotor, and only when using one of the two main industry standard flight controllers (PX4 or ArduPilot). We also corrected some issues with left/right frames being flipped, static transforms published as dynamic and vice versa, and reverted to standard UAS frame conventions where needed for base_link/base_link_frd, map/map_ned, and odom/odom_ned.
+
+### IMU
+
+The best results with respect to IMU seem to come from the mavros topic (`/mavros/imu/data`) rather than the airsim internal IMU topic. To adjust the frequency of MAVROS publishing, change the line in the launch file to:
+
+`<node pkg="rosservice" type="rosservice" name="rosservice" args="call --wait /mavros/set_stream_rate 0 [your_desired_hz] 1"/>`
```

Comment:
I tested 30 and it didn't seem to work. it may be that mavros only implements a subset of the real options (message details [here](http://docs.ros.org/en/hydro/api/mavros/html/srv/StreamRate.html) only list a few). But whatever rate IMU is publishing at, I'd want position data at the same rate, so if there's an option for just those streams, let me know! Otherwise, it doesn't seem to be causing a problem to run "all"/0 (which isn't really "all" but I can't find the details, but e.g. you can check the hz on /mavros/state and it's about 1Hz even when IMU data is at 75Hz)

Reply:
Ah I had the message number wrong, it's 31. You should be able to use '31' for ATTITUDE_QUATERNION mavlink message which in turn publishes `/mavros/imu/data` and '32' for LOCAL_POSE_NED which in turn publishes `/mavros/local_pose/pose`

Reply:
Ah good to know!